src/main/java/fi/otavanopisto/kuntaapi/server/integrations/management/ManagementMenuEntityUpdater.java

```diff
@@ -105,63 +91,34 @@ public void startTimer() {
   }
 
   private void startTimer(int duration) {
-    stopped = false;
     TimerConfig timerConfig = new TimerConfig();
     timerConfig.setPersistent(false);
     timerService.createSingleActionTimer(duration, timerConfig);
   }
-
-  @Override
-  public void stopTimer() {
-    stopped = true;
-  }
   
-  @Asynchronous
-  public void onMenuIdUpdateRequest(@Observes MenuIdUpdateRequest event) {
-    if (!stopped) {
-      OrganizationId organizationId = event.getOrganizationId();
-      
-      if (organizationSettingController.getSettingValue(organizationId, ManagementConsts.ORGANIZATION_SETTING_BASEURL) == null) {
-        return;
-      }
-      
-      queue.add(event);
+  @Timeout
+  public void timeout(Timer timer) {
+    if (systemSettingController.isNotTestingOrTestRunning()) {
+      executeNextTask();
     }
+
+    startTimer(systemSettingController.inTestMode() ? 1000 : TIMER_INTERVAL);
   }
   
-  @Asynchronous
-  public void onMenuIdRemoveRequest(@Observes MenuIdRemoveRequest event) {
-    if (!stopped) {
-      MenuId menuId = event.getId();
-      
-      if (!StringUtils.equals(menuId.getSource(), ManagementConsts.IDENTIFIER_NAME)) {
-        return;
-      }
-      
-      deleteMenu(event.getOrganizationId(), menuId);
-    }
-  }
-
-
-  @Timeout
-  public void timeout(Timer timer) {
-    if (!stopped) {
-      if (systemSettingController.isNotTestingOrTestRunning()) {
-        MenuIdUpdateRequest updateRequest = queue.next();
-        if (updateRequest != null) {
-          updateManagementMenu(updateRequest);
-        }
+  private void executeNextTask() {
+    IdTask<MenuId> task = menuIdTaskQueue.next();
+    if (task != null) {
+      if (task.getOperation() == Operation.UPDATE) {
+        updateManagementMenu(task.getId(), task.getOrderIndex()); 
+      } else if (task.getOperation() == Operation.REMOVE) {
+        deleteManagementMenu(task.getId());
       }
-
-      startTimer(systemSettingController.inTestMode() ? 1000 : TIMER_INTERVAL);
     }
   }
 
-  private void updateManagementMenu(MenuIdUpdateRequest updateRequest) {
-    OrganizationId organizationId = updateRequest.getOrganizationId();
+  private void updateManagementMenu(MenuId managementMenuId, Long orderIndex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "updateManagementMenu" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)